dom/attr/attr-test.js

```diff
@@ -436,6 +436,39 @@ test("Multiselect values is updated on any children added/removed", function(){
 	}
 });
 
+test("Select options within optgroups should be set properly", function() {
+	function tag (tag, value) {
```

Comment:
any reason why you're not using `.innerHTML`?  It might be more immediately clear to someone what sort of structure is being setup.